inbm-lib/inbm_common_lib/shell_runner.py

```diff
@@ -105,15 +107,26 @@ def run_with_log_path(self,
                 shell=False,
                 stdin=subprocess.DEVNULL,
                 stdout=subprocess.PIPE,
-                stderr=subprocess.STDOUT)
+                stderr=subprocess.STDOUT,
+                env=env)
+        elif env:
+            proc = subprocess.Popen(
+                cmd,
```

Comment:
why use `cmd` in block where env is set, but `shlex_split_cmd` otherwise? (also, I notice `env` is passed in to both blocks; if user is not specifying `env` to this function, it should not be passed in to Popen)

Reply:
In the cmd, we provide the `--password $ORAS_PASSWORD`. With `shlex_split_cmd`, it can't read the value of $ORAS_PASSWORD. This is because the $ORAS_PASSWORD is interpreted by the shell and not by Python.

Reply:
Shell runner method updated. If password is provided, the password will be passed as stdin input when running the command.